types/index.d.ts

```diff
@@ -0,0 +1,496 @@
+    constructor(options?: ServerConfig);
+
+    /** The underlying in-memory database instance for this server. */
+    readonly db: Db;
+
+    /** An interface to the Mirage ORM that allows for querying and creating records. */
+    readonly schema: Schema<Registry>;
+
+    /** Creates a model of the given type. */
+    readonly create: Schema<Registry>["create"];
+
+    /** Whether or not Mirage should log all requests/response cycles. */
+    logging: boolean;
+
+    /** A default number of ms to artificially delay responses for all routes. */
+    timing: number;
+
+    /** A default prefix applied to all subsequent route definitions. */
+    namespace: string;
+
+    /** Sets a string to prefix all route handler URLs with. */
+    urlPrefix: string;
+
+    /** Actual Pretender instance */
+    pretender: PretenderServer;
+
+    /** Creates multiple models of the given type. */
+    createList<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<Init>
+    >(modelName: K, count: number, data?: Data): Array<Init & Data>;
+
+    /** Handle a GET request to the given path. */
+    get(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a POST request to the given path. */
+    post(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PUT request to the given path. */
+    put(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PATCH request to the given path. */
+    patch(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle an OPTIONS request to the given path. */
+    options(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a DELETE request to the given path. */
+    del(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    delete(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Pass through one or more URLs to make real requests. */
+    passthrough(urls?: ((request: Request) => any) | string | string[]): void;
+
+    /** Load all available fixture data matching the given name(s). */
+    loadFixtures(...names: string[]): void;
+
+    seeds(server: Server): void;
+
+    routes(): void;
+
+    /** Shutdown the server and stop intercepting network requests. */
+    shutdown(): void;
+  }
+}
+
+declare module "miragejs/db" {
+  import DbCollection from "miragejs/db-collection";
+  import IdentityManager from "miragejs/identity-manager";
+
+  /** The in-memory database containing all currently active data keyed by collection name. */
+  export default class Db {
+    [key: string]: DbCollection | any;
```

Comment:
`| any` here makes the `DbCollection` part meaningless: you're saying the type is "something specific, or anything", which is the same as saying "anything". Likely it should just be `DbCollection`.

Reply:
Removing `any` would cause the following error, any suggestion how could we solve it?

```
Error: /Users/zoltan/projects/miragejs/types/index.d.ts:360:5
ERROR: 360:5  expect  TypeScript@next compile error:
Property 'createCollection' of type '(name: string, initialData?: any[] | undefined) => void' is not assignable to string index type 'DbCollection'.
ERROR: 361:5  expect  TypeScript@next compile error:
Property 'dump' of type '() => void' is not assignable to string index type 'DbCollection'.
ERROR: 362:5  expect  TypeScript@next compile error:
Property 'emptyData' of type '() => void' is not assignable to string index type 'DbCollection'.
ERROR: 363:5  expect  TypeScript@next compile error:
Property 'loadData' of type '(data: any) => void' is not assignable to string index type 'DbCollection'.
```

Reply:
See comment below for how to resolve this – it was a bit of a tricky one!